### ISSUE TITLE
Font in stylesheet is not applied to list

### ISSUE THREAD

**Issue by koendehondt** [guillep/pharo-core#15884]
**Bug description**

I like to apply a dark background color to a list and I like to apply a white font color. I would expect that a stylesheet as in the following code would yield the desired result, but is does not, as shown in the screenshot.

```smalltalk
| box label list application |
label := SpLabelPresenter new
	label: 'Select a number:';
	yourself.
list := SpListPresenter new
	items: (1 to: 10);
	addStyle: 'listBackground';
	yourself.
box := SpBoxLayout newTopToBottom
	add: label expand: false;
	add: list;
	yourself.
application := SpApplication new
	addStyleSheetFromString: '.application [
		.listBackground [ Draw { #color: #16214A }, Font { #color: #FFFFFF } ]
	]';
	yourself.
SpPresenter new
	application: application;
	layout: box; open
```

Note that I clicked on number 6 and that the mouse pointer was hovering over number 9 when I took the screenshot.

<img width="424" alt="How to set the list font color" src="https://github.com/pharo-project/pharo/assets/753271/dfb85f71-a434-47e3-8043-4e6a2a8b23e2">

Did I make a mistake in my code? According to what I read in the code, the stylesheet is correct, so I guess this is a bug.

It is also not clear to me how to apply a different selection color and a different hover color. Does Spec support that?

**Version information:**
- OS: MacOS Sonoma
- Version: 14.1.2
- Pharo Version: 11

**Comment by Ducasse** [comment]
Could you create your application first and add the application to create the presenters? There is a newApp: or something like that.

**Comment by koendehondt** [comment]
@Ducasse 

Do you mean like this?

```smalltalk
| box label list application |
application := SpApplication new
	addStyleSheetFromString: '.application [
		.listBackground [ Draw { #color: #16214A }, Font { #color: #FFFFFF } ]
	]';
	yourself.
label := (SpLabelPresenter newApplication: application)
	label: 'Select a number:';
	yourself.
list := (SpListPresenter newApplication: application)
	items: (1 to: 10);
	addStyle: 'listBackground';
	yourself.
box := SpBoxLayout newTopToBottom
	add: label expand: false;
	add: list;
	yourself.
(SpPresenter newApplication: application)
	layout: box; open
```

It has the same result.

<img width="431" alt="Screenshot 2023-12-30 at 22 58 01" src="https://github.com/pharo-project/pharo/assets/753271/14ae240a-12bf-4896-becb-6b8c73057fd9">

**Comment by koendehondt** [comment]
I did not check all presenters, but it does not work for `SpTextInputFieldPresenter` either. Consider this code:

```smalltalk
| box input |
application := SpApplication new
	addStyleSheetFromString: '.application [
		.textBackground [ Draw { #color: #16214A }, Font { #color: #FFFFFF } ]
	]';
	yourself.
input := (SpTextInputFieldPresenter newApplication: application)
	text: 'Test';
	addStyle: 'textBackground';
	yourself.
box := SpBoxLayout newTopToBottom
	add: input expand: false;
	add: (SpListPresenter newApplication: application);
	yourself.
(SpPresenter newApplication: application)
	layout: box; open
```

At first, it looks like the white color is applied to the text:

<img width="418" alt="Screenshot 2023-12-30 at 23 04 30" src="https://github.com/pharo-project/pharo/assets/753271/7ada53b6-5371-40fe-b458-2a70231fa610">

but let's select the text (all good so far):

<img width="418" alt="Screenshot 2023-12-30 at 23 04 35" src="https://github.com/pharo-project/pharo/assets/753271/00a309d0-0ea6-4681-b138-849f01154da3">

and let's start typing:

<img width="418" alt="Screenshot 2023-12-30 at 23 04 49" src="https://github.com/pharo-project/pharo/assets/753271/7ef1aae3-9af4-45d2-b058-9c6801397dd6">

Now the white text color is gone. The text color is black.

**Comment by Ducasse** [comment]
Yes then I do not know. 
I really hope that soon we will have Toplo (the new widget library) so that people like you can spend time to improve it and let Morphic rest in peace. 
We should have a presentation of the new architecture end of Jan.

**Comment by koendehondt** [comment]
Yes that would be great. I really look forward to Toplo. Do you think the presentation will be available by this month's Pharo Sprint? That would be a good time to talk about it.

**Comment by hernanmd** [comment]
I don't know about Toplo, but this issue could be related to https://github.com/pharo-spec/Spec/issues/1256
So it could be closed here? @estebanlm

**Comment by koendehondt** [comment]
@hernanmd @estebanlm This issue is related to https://github.com/pharo-spec/Spec/issues/1256, but it includes more issues than mentioned there. By closing this issue, that information would be lost. Should I add a comment to https://github.com/pharo-spec/Spec/issues/1256 with the extra issues mentioned here? Or is the link to this issue enough?

**Comment by estebanlm** [comment]
Ultimately, the styling provided by the morphic backend is restrained to the limits of morphic itself.
nothing in morphic allows to directly affect the row. 
this requires an enhancement on the way the style for lists/tables is applied, following what is implemented in `SpMorphicProgressBarAdapter` (which implements some "substyles" for the adapter).
